src/overseer/core.clj

```diff
@@ -1,11 +1,88 @@
 (ns ^:no-doc overseer.core
   "Internal core functions"
   (:require [datomic.api :as d]
-            [clojure.set :as set]))
+            [clojure.set :as set]
+            [miner.herbert :as h]
+            loom.graph))
+
+(def job-schema
+  "Jobs are defined as maps with the requisite descriptive keys/values."
+  '{:job/id str
+    :job/type kw
+    :job/status (or :unstarted :started :finished :failed :aborted)
+    :job/heartbeat int
+    :job/args? map
+    :job/failure? map})
+
+(defn valid-job? [job]
+  (h/conforms? job-schema job))
+
+(defn valid-graph?
+  "Graphs are defined as Loom digraphs graphs where every node is a valid Job.
+  See `loom.graph/digraph`."
+  [graph]
+  (and (satisfies? loom.graph/Digraph graph)
+       (every? valid-job? (loom.graph/nodes graph))))
+
+(defprotocol Store
+  (job-info [this job-id]
+    "Given a job-id, return a Job")
+
+  (transact-graph [this graph]
+    "Given a Graph, atomically transact all of its jobs/dependencies into the store.
+    Side-effecting only; return value is undefined.")
```

Comment:
I think ditto for the rest of the "return value is undefined," I think I would rather report some kind of success/failure, even if the operations are idempotent.